mantle/kola/tests/misc/disk.go

```diff
@@ -0,0 +1,159 @@
+		Platforms:   []string{"qemu-unpriv"},
+	})
+}
+
+func varLibContainers(c cluster.TestCluster) {
+	var m platform.Machine
+	var err error
+	disk := "disk1"
+	varMountName := "var-lib-containers.mount"
+	mkfsServiceName := fmt.Sprintf("systemd-mkfs@dev-disk-by-id-virtio-%s.service", disk)
+
+	options := platform.MachineOptions{
+		AdditionalDisks: []string{"1024M"},
+	}
+
+	mkfsUnit := fmt.Sprintf(`[Unit]
+	Description=Make File System on /dev/disk/by-id/%[1]s
+	DefaultDependencies=no
+	BindsTo=dev-disk-by\x2did-virtio\x2d%[1]s.device
+	After=dev-disk-by\x2did-virtio\x2d%[1]s.device var.mount
+	Before=systemd-fsck@dev-disk-by\x2did-virtio\x2d%[1]s.service
+	Before=shutdown.target
+
+	[Service]
+	Type=oneshot
+	RemainAfterExit=yes
+	ExecStart=/bin/bash -c "/bin/rm -rf /var/lib/containers/*"
+	ExecStart=mkfs.xfs /dev/disk/by-id/virtio-%[1]s
+	TimeoutSec=0
+
+	[Install]
+	WantedBy=var-lib-containers.mount`, disk)
+
+	varMountUnit := fmt.Sprintf(`[Unit]
+	Description=Mount %[2]s disk to /var/lib/containers
+	Before=local-fs.target
+	Requires=%[1]s
+	After=%[1]s
+
+	[Mount]
+	What=/dev/disk/by-id/virtio-%[2]s
+	Where=/var/lib/containers
+	Type=xfs
+	Options=defaults,prjquota
+
+	[Install]
+	WantedBy=local-fs.target`, mkfsServiceName, disk)
+
+	config := ignv3types.Config{
+		Ignition: ignv3types.Ignition{
+			Version: "3.0.0",
+		},
+		Storage: ignv3types.Storage{
+			Disks: []ignv3types.Disk{
+				{
+					Device:    "/dev/disk/by-id/virtio-disk1",
+					WipeTable: util.BoolToPtr(true),
+				},
+			},
+			Files: []ignv3types.File{
+				{
+					Node: ignv3types.Node{
+						Path: "/var/home/core/Dockerfile",
+					},
+					FileEmbedded1: ignv3types.FileEmbedded1{
+						Contents: ignv3types.FileContents{
+							Source: util.StrToPtr("data:,FROM%20scratch%0ACOPY%20os-release%20%2F"),
+						},
+						Mode: util.IntToPtr(420),
+					},
+				},
+			},
+		},
+		Systemd: ignv3types.Systemd{
+			Units: []ignv3types.Unit{
+				{
+					Name:     mkfsServiceName,
+					Contents: &mkfsUnit,
+					Enabled:  util.BoolToPtr(true),
+				},
+				{
+					Name:     varMountName,
+					Contents: &varMountUnit,
+					Enabled:  util.BoolToPtr(false),
+				},
+			},
+		},
+	}
+
+	buf, err := json.Marshal(config)
+	if err != nil {
+		c.Fatalf("Unable to Marshal config")
+	}
+	serializedConfig := buf
+
+	m, err = c.NewMachineWithOptions(conf.Ignition(string(serializedConfig)), options)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Verify /var/lib/containers isn't mounted then enable the units to move
+	// it to the secondary disk
+	vlc := "/var/lib/containers"
+	if mount, err := c.SSH(m, fmt.Sprintf("findmnt %s -n -o SOURCE", vlc)); err == nil {
+		c.Fatalf("%s is already mounted on %s", vlc, string(mount))
+	}
+
+	// build a small container image to exercise /var/lib/container storage
+	c.MustSSH(m, "cp /etc/os-release /var/home/core")
+	c.MustSSH(m, "sudo podman build -t test -f Dockerfile .")
+	c.MustSSH(m, fmt.Sprintf("sudo systemctl enable %s", mkfsServiceName))
```

Comment:
This is already enabled in the Ignition config, is something disabling it?

Reply:
Good catch.  It should be disabled in the ignition config.